jetty-core/jetty-io/src/test/java/org/eclipse/jetty/io/ByteArrayEndPointTest.java

```diff
@@ -270,33 +265,6 @@ public void testIdle() throws Exception
 
         assertTrue(endp.isOpen());
         Thread.sleep(oneAndHalfIdleTimeout);
-        // Still open because it has not been oshut or closed explicitly
-        // and there are no callbacks, so idle timeout is ignored.
-        assertTrue(endp.isOpen());
-
-        // Normal read is immediate, since there is data to read.
-        ByteBuffer buffer = BufferUtil.allocate(1024);
-        FutureCallback fcb = new FutureCallback();
-        endp.fillInterested(fcb);
-        fcb.get(idleTimeout, TimeUnit.MILLISECONDS);
-        assertTrue(fcb.isDone());
-        assertEquals(4, endp.fill(buffer));
-        assertEquals("test", BufferUtil.toString(buffer));
-
-        // Wait for a read timeout.
-        long start = NanoTime.now();
-        fcb = new FutureCallback();
-        endp.fillInterested(fcb);
-        try
-        {
-            fcb.get();
-            fail("Expected ExecutionException");
-        }
-        catch (ExecutionException t)
-        {
-            assertThat(t.getCause(), instanceOf(TimeoutException.class));
-        }
-        assertThat(NanoTime.millisSince(start), greaterThan(halfIdleTimeout));
-        assertThat("Endpoint open", endp.isOpen(), is(true));
+        assertFalse(endp.isOpen());
```

Comment:
Why is the rest of the test removed?

Reply:
Because we removed the logic in AbstractEndPoint about only closing on idle if we were already half closed.